src/utils/globalScript.ts

```diff
@@ -7,7 +7,11 @@ import { initThemeChangeEvent } from "./theme";
  */
 export default function (): void {
   if (isBrowser()) {
-    document.addEventListener("DOMContentLoaded", () => initThemeChangeEvent());
+    if (document.readyState === "interactive") {
+      initThemeChangeEvent();
+    } else {
+      document.addEventListener("DOMContentLoaded", () => initThemeChangeEvent());
```

Comment:
Can you use the `once` argument so the listener is removed after the event is handled?